test/libsolidity/util/TestFileParser.cpp

```diff
@@ -72,6 +72,19 @@ namespace
 	}
 }
 
+char TestFileParser::Scanner::peek() const {
+	if (std::distance(m_char, m_line.end()) < 2)
+		return '\0';
+
+#if __cplusplus >= 201703L
```

Comment:
I really don't think this is worth the hassle...

What about `*m_char++`?

Reply:
`*m_char++` is also modifying `m_char`, which wasn't meant to. That's why the hassle of using a temporary. Or - the more concise version with `std::next()`, which however isn't available pre-C++17.

I think I provided the `#ifdef` only because I prefer the first part of the solution.